Fix links track with use_middle

Jeff1995 opened the pull request:
Proposed bug fix for #422.

lldelisle commented:
Thank you very much for the bug fix, indeed, I did not think about this case.
What is missing to this PR is:
- target to develop instead of master and rebase or merge develop
- add a test so we never reintroduce this bug
- add a release text to announce this bug fix
Tell me if you are too busy to do it, I can do it as well.

Jeff1995 commented:
Sure! I have rebased and targeted to develop, and modified a test file to include a situation that would have raised the error. Release text has also been added. Note that the modifed test file "test_wide.arcs" causes change to three test images. Could you check and see if that's appropriate?

lldelisle commented:
The test looks good thank you. I am just wondering if the output of master_arcs_use_middle.png should not be changed to show like we see on master_links_squares_overlay.png...

Jeff1995 commented:
I guess this is just a corner case where the triangular and square visualizations look awkward, and should not have been used in the first place.

lldelisle commented:
Yes, I think you are right, we never thought that end2 could be smaller than end 1....
I guess the expected behaviour would be to go from start1 to max(end1, end2)

lldelisle commented:
What I was mentioning is what I draw in pink which is missing:
![image](https://user-images.githubusercontent.com/30404086/205085380-362eeefc-1ca5-44f7-9110-61fa9f2ddc9c.png)
Also because we never thought about the case end2>end1

Jeff1995 commented:
Thanks! Now I understand 😀. I agree with these expected behaviors. I'll see what I can do and get back to you tomorrow.

lldelisle commented:
Thank you very much. If you don't have time, I can do it.

Jeff1995 commented:
I think I've got it to work.

<img width="856" alt="Screen Shot 2022-12-02 at 14 22 06" src="https://user-images.githubusercontent.com/13842130/205228743-234c1934-e429-4bc0-8bcb-f397e8789db3.png">

lldelisle commented:
This is super great! Thank you so much.

lldelisle commented on docs/content/releases/3.7.1.rst:
```diff
@@ -5,4 +5,4 @@ Bugfix:
 
 - Works with gffutils version 0.11.0 where ValueError became gffutils.exceptions.EmptyInputError for empty gtf.
-
+- Fixed a small bug in links track with "use_middle" enabled, where midpoints could have different ordering than start points.
```
```suggestion
- Fixed a small bug in links track with "use_middle" enabled, where midpoints could have different ordering than start points (Thanks to @Jeff1995 ).
```
